coalib/results/result_actions/OpenEditorAction.py

```diff
@@ -39,23 +99,44 @@ def apply(self, result, original_file_dict, file_diff_dict, editor: str):
 
         :param editor: The editor to open the file with.
         """
-        # Use set to remove duplicates
-        filenames = {src.file: src.renamed_file(file_diff_dict)
-                     for src in result.affected_code}
-
-        editor_args = [editor] + list(filenames.values())
-        arg = EDITOR_ARGS.get(editor.strip(), None)
-        if arg:
-            editor_args.append(arg)
-
-        # Dear user, you wanted an editor, so you get it. But do you really
-        # think you can do better than we?
-        if editor in GUI_EDITORS:
-            subprocess.call(editor_args, stdout=subprocess.PIPE)
+        if editor in KNOWN_EDITORS:
+            editor_info = KNOWN_EDITORS[editor.strip()]
+        else:
+            # If the editor is unknown fall back to just passing
+            # the filenames
+            editor_info = {
+                'file_arg_template': '{filename}',
+                'gui': True
+            }
+
+        # Use dict to remove duplicates
+        filenames = {
+            src.file: {
+                'filename': src.renamed_file(file_diff_dict),
+                'line': src.start.line,
+                'column': src.start.column
+            }
+            for src in result.affected_code
+        }
+
+        call_args = [editor]
+
+        # for some editors we define extra arguments
+        if 'args' in editor_info:
+            call_args.append(editor_info['args'])
+
+        # add info for each file to be opened
+        for file_info in filenames.values():
+            file_arg = editor_info['file_arg_template'].format(**file_info)
+            call_args += file_arg.split(' ')
+
+        if editor_info.get('gui', True):
+            subprocess.call(call_args, stdout=subprocess.PIPE)
         else:
-            subprocess.call(editor_args)
+            subprocess.call(call_args)
```

Comment:
can we put this whole arg creation logic into an own function? That's also easier to test without mocking everything, just a function that takes the editor and gives back args and kwargs to the subprocess (or sarge?) call